### ISSUE TITLE
Enable passing context to key store services

### ISSUE THREAD

**Issue by csatib02** [bank-vaults/bank-vaults#2893]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/bank-vaults/bank-vaults/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://bank-vaults.dev/docs/support) without success.
- [X] I agree to follow the [Code of Conduct](https://bank-vaults.dev/docs/code-of-conduct).

### Version

v1.31.1

### Expected Behavior

Currently key-store services don't accept context as parameter.

### Actual Behavior

Enable passing context to key-store Set/Get functions.

**Comment by sureshkrishnan-v** [comment]
can i work on this issue i am new contributer with 2 yr of go experience

**Comment by csatib02** [comment]
Hey @sureshkrishnan-v,
Absolutely, thank you!

**Comment by sureshkrishnan-v** [comment]
Hey, thank you! Can you please describe this issue? I’m a new contributor, and it will help me understand and create a PR

like we have many key/value services aws , azure so i need to support for all

**Comment by csatib02** [comment]
Basically we would want to support all key-value store's getters and setters to use a context parameter.

**Comment by sureshkrishnan-v** [comment]
yep i have seen that here should i need to pass ctx as timeout or else background 



// Service defines a basic key-value store. Implementations of this interface
// may or may not guarantee consistency or security properties.
type Service interface {
	Set(ctx context.Context, key string, value []byte) error
	Get(ctx context.Context, key string) ([]byte, error)
}


type Vault interface {
	Init(ctx context.Context) error
	RaftInitialized(ctx context.Context) (bool, error)
	RaftJoin(leaderAddress string) error
	Sealed() (bool, error)
	Active() (bool, error)
	Unseal(ctx context.Context) error
	Leader() (bool, error)
	LeaderAddress() (string, error)
	Configure(ctx context.Context, config map[string]interface{}) error
}

type KVService interface {
	Set(ctx context.Context, key string, value []byte) error
	Get(ctx context.Context, key string) ([]byte, error)
}
